--- a/tstore/utils/src/common/OracleConnection.cc
+++ b/tstore/utils/src/common/OracleConnection.cc
@@ -172,7 +172,7 @@
 template<class T>
 std::string integerColumnType(const std::string &columnName) {
 	std::ostringstream intType;
-	unsigned int digits=(unsigned int)ceil(log10(std::numeric_limits<T>::max())); //std::numeric_limits<T>::digits10  seems to give one less than it should
+	unsigned int digits=(unsigned int)ceil(log10(static_cast<double>(std::numeric_limits<T>::max()))); //std::numeric_limits<T>::digits10  seems to give one less than it should
 	intType << "NUMBER(" << digits << ") CHECK (" << integerCheckConstraint<T>(columnName)+")";
 	return intType.str();
 }
