--- a/amc13/Makefile
+++ b/amc13/Makefile
@@ -90,6 +90,12 @@
 -include $(OBJECT_FILES:.o=.d)
 -include $(LIBRARY_OBJECT_FILES_UNPACKER:.o=.d)
 
+install:
+	mkdir -p ${INSTALL_PREFIX}/{include,lib,name}
+	find ./include \( -name "*.[hi]" -o -name "*.hpp" -o -name "*.hh" -o -name "*.hxx" \) -exec install -D -m 644 {} ${INSTALL_PREFIX}/{} \;
+	find ./lib -name ".svn" -prune -o -exec install -D -m 755 {} ${INSTALL_PREFIX}/{} \;
+	find ./etc -name ".svn" -prune -o -exec install -D -m 644 {} ${INSTALL_PREFIX}/{} \;
+
 rpm: _rpmall
 _rpmall: 
 	mkdir -p ${RPMBUILD_DIR}/{RPMS/{i386,i586,i686,x86_64},SPECS,BUILD,SOURCES,SRPMS}
--- a/Makefile
+++ b/Makefile
@@ -7,7 +7,7 @@
 
 FLAGS = $(ifeq $(MAKEFLAGS) "","",-$(MAKEFLAGS))
 
-TARGETS=clean cleanrpm rpm build all
+TARGETS=clean cleanrpm rpm build all install
 
 .PHONY: $(TARGETS)
 default: build
--- a/python/Makefile
+++ b/python/Makefile
@@ -41,6 +41,8 @@
 	$(AMC13_ROOT)/amc13/lib \
 	$(CACTUS_ROOT)/lib 
 
-
 include ${AMC13_ROOT}/config/mfRules.mk
 include ${AMC13_ROOT}/config/mfPythonRPMRules.mk
+
+install:
+
--- a/tools/Makefile
+++ b/tools/Makefile
@@ -99,6 +99,15 @@
 
 -include $(LIBRARY_OBJECT_FILES:.o=.d)
 
+install:
+	# create the folder structure
+	mkdir -p ${INSTALL_PREFIX}/bin/amc13/
+	mkdir -p ${INSTALL_PREFIX}/lib/
+	
+	# copy files over
+	find ./bin -name ".svn" -prune -o -exec install -D -m 755 -t ${INSTALL_PREFIX}/bin/amc13 {} \;
+	find ./lib -name ".svn" -prune -o -exec install -D -m 755 {} ${INSTALL_PREFIX}/{} \;
+	
 rpm: _rpmall
 _rpmall: 
 	# create the folder structure
